src/gentropy/method/locus_breaker_clumping.py

```diff
@@ -12,103 +11,163 @@
+                )
+                .withColumn("rank", f.rank().over(w3))
+                .filter(
+                    (f.col("rank") == 1) & (f.col("negLogPValue") > neglog_pv_cutoff)
+                )
+                .select(
+                    *columns_sumstats_columns,
+                    # To make sure that the type of locusStart and locusEnd follows schema of StudyLocus:
+                    f.col("locusStart").cast(t.IntegerType()).alias("locusStart"),
+                    f.col("locusEnd").cast(t.IntegerType()).alias("locusEnd"),
+                    f.lit(None)
+                    .cast(t.ArrayType(t.StringType()))
+                    .alias("qualityControls"),
+                    StudyLocus.assign_study_locus_id(
+                        f.col("studyId"), f.col("variantId")
+                    ).alias("studyLocusId"),
+                )
+            ),
+            _schema=StudyLocus.get_schema(),
+        )
+
+    @staticmethod
+    def process_locus_breaker_output(
+        study_locus: StudyLocus,
+        sum_stats: SummaryStatistics,
+        large_loci_size: int,
+        wbc_clump_distance: int,
+        gwas_threshold: float,
+    ) -> StudyLocus:
+        """Process the locus breaker method result, and run window-based clumping on large loci.
+
+        Args:
+            study_locus (StudyLocus): StudyLocus object with locus start and end positions.
+            sum_stats (SummaryStatistics): Input summary statistics dataset.
+            large_loci_size (int): the size to define large loci which should be broken with wbc.
+            wbc_clump_distance (int): Clump distance for window-based clumping.
+            gwas_threshold (float): P-value threshold to be used in window-based clumping.
+
+        Returns:
+            StudyLocus: clumped study loci with large loci broken by window-based clumping.
+        """
+        small_loci = study_locus.filter(
+            (f.col("locusEnd") - f.col("locusStart")) <= large_loci_size
+        )
+        large_loci = study_locus.filter(
+            (f.col("locusEnd") - f.col("locusStart")) > large_loci_size
+        )
+        large_loci_ss = SummaryStatistics(
+            sum_stats.df.alias("ss")
+            .join(
+                large_loci.df.alias("ll"),
+                (f.col("ss.studyId") == f.col("ll.studyId"))
+                & (f.col("ss.chromosome") == f.col("ll.chromosome"))
+                & (f.col("ss.position") >= f.col("ll.locusStart"))
+                & (f.col("ss.position") <= f.col("ll.locusEnd")),
+                "inner",
             )
-        ),
-        _schema=StudyLocus.get_schema(),
-    )
+            .select([f.col("ss." + col) for col in sum_stats.df.columns]),
+            SummaryStatistics.get_schema(),
+        )
+        large_loci_wbc = WindowBasedClumping.clump(
+            large_loci_ss, wbc_clump_distance, gwas_threshold
+        ).df.withColumns(
```

Comment:
Also, shouldn't we just add this feature to the window based clumping function? So after any clumping, we'll get the boundaries automatically?

Reply:
That is originally how I implemented it, however Yakov pointed out we are using 500kb as the distance variable for windows-based clumping of the large loci, but the actual loci size for these clumps should still be +- 750kb. This logic is specific to the locus-breaker class so I've moved it here, but there is probably a more elegant solution.

Reply:
OK.